sklearn/tests/test_common.py

```diff
@@ -103,6 +103,34 @@ def _sample_func(x, y=1):
     pass
 
 
+class CallableEstimator:
+    """Dummy development stub for an estimator.
+
+    This is to make sure a callable estimator passes common tests.
+    """
+
+    def get_params(self, *, deep=True):
+        return {}
+
+    def set_params(self, **kwargs):
+        pass
+
+    def fit(self, X, y=None):
+        return self
+
+    def fit_transform(self, X, y=None):
+        return self.fit(X, y).transform(X)
+
+    def transform(self, X):
+        return X
+
+    def __call__(self, X):
+        return self.transform(X)
+
+    def __str__(self):
+        return type(self).__name__
+
+
```

Comment:
No need to implement all these methods for this test. We just need `__call__`, `get_params` and ``__str__``, and BaseEstimator alread provides 2 of them. I would just replace it by
```py
class CallableEstimator(BaseEstimator):
    """Dummy development stub for an estimator.

    This is to make sure a callable estimator passes common tests.
    """

    def __call__(self):
        pass
```
Then the expected id would be ``CallableEstimator()`` due to the implementation of str in BaseEstimator.